gato/tasks/caption_task.py

```diff
@@ -0,0 +1,174 @@
+# Assume all datasets are downloaded and available from local directories
+from gato.tasks.task import Task, TaskTypeEnum
+
+import os
+import tarfile
+import webdataset as wds
+import fnmatch
+from torch.utils.data import DataLoader
+from PIL import Image
+import io # need to use BytesIO
+
+import numpy as np
+import math
+import torch
+from torch.nn import functional as F
+from torch import nn
+import json
+import random
+from transformers import AutoTokenizer, GPT2Tokenizer
+
+class CaptionTask(Task): 
+    def __init__(self, task_type: TaskTypeEnum, tokenizer_model:str, caption_dataset, train_data, test_data = [],
+                 test_data_prop = 0.1, test_data_mask_file = None):
+        """
+        task_type should be CAPTION
+        caption_dataset is the directory for all of the data (training and test)
+        train_data and test_data are list of sub_diretories under caption_dataset, with each directory containing 
+        one training or test dataset which is composed of multiple .tar files downloaded with img2dataset. 
+        Each tar file contains multiple bundles, with each bundle containing one .jpg, one txt and one json file. 
```

Comment:
We should to have a better standard here, so as we scale up machines, it's easy to try a different dataset.